src/reporting.ts

```diff
@@ -56,18 +54,20 @@ export const getNotesInMarkdown = (): string => {
 };
 
 export const generateMarkdownReport = (): void => {
-    const workspaceFolder = vscode.workspace.rootPath;
-    // TODO: What if there's no workspace?
-    if (workspaceFolder) {
-        // TODO: Remove this hardcoded string, it should be a configuration
-        const extensionDirPath = path.join(workspaceFolder, '.vscode', 'code-annotation');
-        const extensionFilePath = path.join(extensionDirPath, 'summary.md');
-        let content = getNotesInMarkdown();
-        fs.writeFileSync(extensionFilePath, content);
-        var openPath = vscode.Uri.file(extensionFilePath);
-        vscode.workspace.openTextDocument(openPath).then(doc => {
-            vscode.window.showTextDocument(doc).then(editor => {
-            });
+    const newFile = vscode.Uri.parse('untitled:summary.md');
+
+    vscode.workspace.openTextDocument(newFile).then(summaryFile => {
+        const edit = new vscode.WorkspaceEdit();
+        let notesSummary = getNotesInMarkdown();
+
+        edit.insert(newFile, new vscode.Position(0, 0), notesSummary);
+
+        return vscode.workspace.applyEdit(edit).then(success => {
+            if (success) {
+                vscode.window.showTextDocument(summaryFile);
+            } else {
+                vscode.window.showInformationMessage('Error: Code Annotation could not generate a summary');
```

Comment:
@thamara not sure this error message is enough. 

I also though that it might make sense to choose the wording between `summary` and `report`, as it seems a bit confusing at the moment since both are used meaning the same thing.

Reply:
I think it's enough. It's not completely clear to me what are the scenarios where this happens, but I think just by having this is enough for now.

Reply:
About the report/summary, it makes complete sense, and I think `summary` is the best option as it's the name of the feature.